import/PopupStackView.qml

```diff
@@ -0,0 +1,195 @@
+ * PopupManager, or any of their properties, the popupManagers must be
+ * re-set to the PopupStackView.
+ */
+Control {
+    id: popupStackView
+
+    /*!
+     * \brief A list of PopupManagers that controls the information being 
+     * displayed in a PopupStackView.
+     * \qmlproperty list<PopupManager> popupManagers
+     */
+    property var popupManagers: null
+
+    /*!
+     * \brief This property holds the current top-most item in the stack.
+     * \qmlproperty PopupView currentItem
+     * \sa Stack
+     */
+    property alias currentItem: stack.currentItem
+
+    /*!
+     * \brief This property holds the number of items currently pushed onto the 
+     * stack.
+     * \qmlproperty int depth
+     * \sa Stack
+     */
+    property alias depth: stack.depth
+
+    /*!
+     * \brief This property holds whether a transition is running.
+     * \qmlproperty bool busy
+     * \sa Stack
+     */
+    property alias busy: stack.busy
+
+    /*!
+     * \brief This property holds the transition that is applied to the item 
+     * that enters the stack when another item is popped off of it.
+     * \qmlproperty Transition popEnter
+     * \sa Stack
+     */
+    property alias popEnter: stack.popEnter
+
+    /*!
+     * \brief This property holds the transition that is applied to the item 
+     * that exits the stack when the item is popped off of it.
+     * \qmlproperty Transition popExit
+     * \sa Stack
+     */
+    property alias popExit: stack.popExit
+
+    /*!
+     * \brief This property holds the transition that is applied to the item 
+     * that enters the stack when the item is pushed onto it.
+     * \qmlproperty Transition pushEnter
+     * \sa Stack
+     */
+    property alias pushEnter: stack.pushEnter
+
+    /*!
+     * \brief This property holds the transition that is applied to the item 
+     * that exits the stack when another item is pushed onto it.
+     * \qmlproperty Transition pushExit
+     * \sa Stack
+     */
+    property alias pushExit: stack.pushExit
+
+    /*!
+     * \brief Signal emitted when an attachment thumbnail is clicked.
+     * The \a index of the PopupAttachment in the PopupAttachmentListModel
+     * of the currently displayed PopupView.
+     */
+    signal attachmentThumbnailClicked(var index)
+
+    clip: true
+
+    implicitWidth: 300 + padding
+
+    implicitHeight: 300 + padding
+
+    padding: 5
+
+    background: Rectangle {
+        color: palette.base
+        border {
+            color: palette.shadow
+            width: 2
+        }
+        radius: 2
+    }
+
+    contentItem: GridLayout {
+        columns: 2
+        anchors {
+            fill: parent
+            margins: 5
+        }
+        Button {
+            text: "Previous"
+            onClicked: stack.pop()
+            Layout.alignment: Qt.AlignLeft
+            Layout.fillWidth: true
+            enabled: popupManagers ? stack.depth > 1 : false
```

Comment:
so, will the stack just cycle round when you reach the first/bottom entry? If not I think it needs to be something like:

```suggestion
            enabled: popupManagers ? (stack.depth > 1 && stack.currentIndex > 0) : false
```

Reply:
This will work as written, depth is variable!